### ISSUE TITLE
Plex Update

### ISSUE THREAD

**Issue by hessercan** [cjmurph/PmsService#45]
Are any updates being worked on to address the update issue? It shouldn't be hard for the updater to be automatically run as administrator without the UAC prompt.

**Comment by cjmurph** [comment]
If by update issue you mean you shouldn't use the updates via the plex web interface if you use this service... the anwser is no.

Years ago I started speaking to the plex guys about looking for a specific exit code so I could know it was an update and not auto restart the service. Then the plex forum hack happened and we never got back to it.

This service specifically does not require admin rights, therefore, it won't be installing anything.

**Comment by d8ahazard** [comment]
@cjmurph , @hessercan - Almost got a fix for this one worked out too.

By using the registry to find the pms data path, then setting a filewatcher on the log for the plex update service, it's pretty easy to determine when Plex has stopped itself to install an update - then wait for the log to indicate it's been restarted, then kill the process and grab it again with the service. ;)

Just testing it out now to ensure I didn't do anything dumb, and that'll be pushed to my fork of this till murph and I can chat a bit.

**Comment by TechieGuy12** [comment]
Just curious which registry keybyou are using? The LocalAppData value in the Plex registry key isn't reliable as not all installs have the key. That key isn't needed to run Plex.

**Comment by d8ahazard** [comment]
> Just curious which registry key you are using? The LocalAppData value in the Plex registry key isn't reliable as not all installs have the key. That key isn't needed to run Plex.

Good point!

That is the key I was looking for, and given your suggestion, I've added a bit to the helper that starts by looking in %localappdata%\... (default) location, and then checking in the registry if it's not in the default position.

![image](https://user-images.githubusercontent.com/1633844/145143821-f96efbe4-c870-46dc-9c60-37c03fe235bb.png)

**Comment by TechieGuy12** [comment]
That should work. I believe the LocalAppDataPath registry key doesn't include the Plex Media Server folder, so if you are returning that folder, as you do with the first Environment.SpecialFolder.LocalApplicationData check, you will need to append "Plex Media Server" to the registry key value.

**Comment by d8ahazard** [comment]
Noted and done. 😄

**Comment by arrmo** [comment]
Sorry, trying to understand this one - and not fully getting it 😞. Is this the case where PMS is stuck in the state of `Plex updating`?

Thanks!

**Comment by cjmurph** [comment]
> Sorry, trying to understand this one - and not fully getting it 😞. Is this the case where PMS is stuck in the state of `Plex updating`?
> 
> Thanks!

no, this issue is for the plex service to handle updates at all. 
I see why it would get stuck in updating, I noticed last night while testing that while one update worked, another seemed to require user input and since the update was happening in session 0, that wasn't possible. In that case, plex was stuck "updating" and in fact will not update and will not restart. That is in fact a seperate issue.

**Comment by arrmo** [comment]
Meaning the update was waiting for user input => and it's hidden, so can't see it, nor respond. Correct? The problem in that case is no real way to restart it?

Thanks!

**Comment by cjmurph** [comment]
It seems like it.

**Comment by arrmo** [comment]
That makes sense - and no real way around it I assume?

Thanks!
